target/ppc/translate/vmx-impl.c.inc

```diff
@@ -3110,6 +3106,52 @@ TRANS_FLAGS2(ALTIVEC_207, VPMSUMD, do_vx_helper, gen_helper_VPMSUMD)
 TRANS_FLAGS2(ALTIVEC_207, VSUBCUQ, do_vx_helper, gen_helper_VSUBCUQ)
 TRANS_FLAGS2(ALTIVEC_207, VSUBUQM, do_vx_helper, gen_helper_VSUBUQM)
 
+static void gen_VADDCUW(unsigned vece, TCGv_vec t, TCGv_vec a, TCGv_vec b)
+{
+    tcg_gen_not_vec(vece, a, a);
+    tcg_gen_cmp_vec(TCG_COND_LTU, vece, t, a, b);
+    tcg_gen_neg_vec(vece, t, t);
+}
```

Comment:
The helper was `r->u32[i] = ~a->u32[i] < b->u32[i];` but this now looks like `~(~a < b)`. Is this right or am I missing something?

Reply:
neg != not, this is `-(~a < b)` and that's because in C when `(~a < b)` is true it returns 1, but gvec inline operations return -1 (aka all bits are 1), so at the end negate the value (`-(-1)`) to get the same result

Reply:
Yes indeed, my bad. This is correct, thanks for the explanation

Reply:
Maybe
```c
TCGv_vec one = tcg_constant_vec_matching(t, vece, 1);
tcg_gen_not_vec(vece, t, a);
tcg_gen_cmp_vec(TCG_COND_LTU, vece, t, t, b);
tcg_gen_and_vec(t, t, one);
```
To avoid the optional `INDEX_op_neg_vec` opcode